### PR TITLE
Refactor to allow set_config and templater to be consistent.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Description**

Sets aside common utils for use in the CLI tools. Uses the same naming convention for the input arguments as templater, so requires some checks and changes to the tests to accommodate that modification.

This PR should be merged after #203 as it is higher priority, and this one will have some significant merge conflicts.

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

Co-authored-by: @j-derrico

### CODE REVIEW

File: src/uwtools/utils/cli_helpers.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+def path_if_file_exists(arg):
+    ''' Checks whether a file exists, and returns the path if it does. '''
+    if not os.path.exists(arg):
+        msg = f'{arg} does not exist!'
+        raise argparse.ArgumentTypeError(msg)
+
+    return os.path.abspath(arg)
+
+def setup_logging(user_args, logfile=None):
```
Comment (WeirAE) [inline src/uwtools/utils/cli_helpers.py:43]: Ah, this should probably be modified to reflect the #203 changes just to make sure it doesn't get lost in the significant refactoring here

File: tests/test_set_config.py [hunk 2]
```diff
@@ -40,11 +42,11 @@ def test_path_if_file_exists(): #pylint: disable=unused-variable
     non-existant path.'''
 
     with tempfile.NamedTemporaryFile(dir='.', mode='w') as tmp_file:
-        assert set_config.path_if_file_exists(tmp_file.name)
+        assert cli_helpers.path_if_file_exists(tmp_file.name)
 
     with pytest.raises(argparse.ArgumentTypeError):
         not_a_filepath = './no_way_this_file_exists.nope'
-        set_config.path_if_file_exists(not_a_filepath)
+        cli_helpers.path_if_file_exists(not_a_filepath)
```
Comment (WeirAE) [inline tests/test_set_config.py:49]: This should probably go to a test_cli_helpers.py when we get to refactoring the tests

**Comment by venitahagerty** [comment]
Did one run each for set_config and templater. Ran well, got expected results. Looks good except for updating logging.

**Comment by christinaholtNOAA** [comment]
@venitahagerty @WeirAE were you okay with this PR? Would you mind pushing the "Accept" button, if so?

**Comment by venitahagerty** [comment]
What's the plan for updating logging? In this PR, or in another issue?

**Review by WeirAE — APPROVED** [review]
Sorry,  did not see the update message, but LGTM! log name and stack inspect look properly incorporated

**Comment by christinaholtNOAA** [comment]
OMG! @venitahagerty I think maybe I totally stopped reading at "looks good". 😬 Sorry.

Do you mind elaborating on what was wrong with logging?

**Comment by venitahagerty** [comment]
Brian had commented about updating the logging in cli_helpers.py to match UW 203. Was that resolved?

**Comment by WeirAE** [comment]
Yep, it got a little squirrely in there, but it looks like its all in the last commit that for some reason I had missed the notification for!

**Review by venitahagerty — APPROVED** [review]
LGTM

**Comment by christinaholtNOAA** [comment]
Ahhh! Oh, yes. Thanks for the clarification. I believe I finally got it squared away, and even added a logfile option to set_config so that is now consistent with templater, too.

**Pull request merged**